### PR TITLE
Fix/TR-3995/Max words restriction can be exceeded

### PR DESCRIPTION
**Pull request by jsconan**
1/4

Related to: https://oat-sa.atlassian.net/browse/TR-3995

The helper `util/strLimiter` is aimed at limiting a text to a maximum number of units, either chars or words. Unfortunately, so far it was exclusively expecting plain text and it was producing wrong results with HTML.

This pull request proposes to support either plain text or HTML text.

How to test:
- run the unit tests
- create a test having extended text interactions with a constraint for the max words and for the max length (a test sample is attached to the ticket).
- publish the test, then take it
- type some text in the extended text interaction, then copy it and paste it until the limit is reached.
- try to exhaust the limit by pasting again
- remove a few words, ideally less than the number in the clipboard, then paste again

Without the fix, the limit can be overflown.
With the fix, we cannot overflow the limit.

### CODE REVIEW

**Review by meefox — APPROVED** [review]
Screenshot 2022-05-11 at 17 43 12

**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [ ] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] Commits messages are meaningful
- [ ] Commits are atomic
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Review by shaveko — APPROVED** [review]
Code looks fine and tests run well, please check my comment on optional change

File: src/util/strLimiter.js [hunk 1]
```diff
@@ -13,10 +13,50 @@
...
+ * @param {function} limitText
+ * @returns {string}
+ */
+function limitBy(text, limitText) {
+    /**
+     * Limits the size of an HTML fragment, removing the extraneous content.
+     * @param {Node} fragment
+     */
+    const limitFragment = fragment => {
+        [].slice.call(fragment.childNodes).forEach(node => {
```
[the remark this answers is not in the document]
Comment (jsconan) [reply src/util/strLimiter.js:31]: Indeed, I was considering using it. But as there might be a need to backport it to an older version than needs IE support I preferred to stay safe.

**Comment by AntonNeviarouski** [comment]
QA verified

**Pull request merged**